### ISSUE TITLE
Clear inline functions

### ISSUE THREAD

**Issue by TheJaredWilcurt** [tjw-lint/jest-serializer-vue-tjw#74]
Add `clearInlineFunctions` to the API and default to false. Use regex to detect if an attribute's value matches the pattern of a function, then replace the value with a placeholder like `[function]`.

Spinoff of #71.

**Comment by cameron-martin** [comment]
Couldn't this be handled by modifying the vnode prior to serialisation, rather than as a post-processing step? That way no regex magic would have to be used.

It seems like you do a similar thing for serialisation of object properties.

**Comment by TheJaredWilcurt** [comment]
Any VNode related changes are experimental. Because a test can have multiple expectations. If the first one is a `expect(wrapper).toMatchSnapshot()` and we mutate the VNode, then it may effect any other `expect` that comes after. So to avoid this we deep clone the VNode object. However, depending on how complex the component is, the VNode may be extremely large and cause the callstack to be exceeded, forcing the test to fail and throw an error.

So whenever possible, if we can add a feature without having to use VNodes, it is much safer. Ultimately the result needs to be returned as a string anyway, so doing direct string manipulation is fine, and regex comes with that turf.

**Comment by cameron-martin** [comment]
Instead of deep cloning the whole tree you could instead only create new copies of the vnodes that need to be modified and their ancestors. This way in the case that no vnodes need to be modified then no cloning happens.

**Comment by TheJaredWilcurt** [comment]
@cameron-martin I'm not sure what you mean, If you want to do a proof-of-concept PR I can take a look. But I hesitate to spend any time doing Vnode specific stuff, since Vue 3 is so different from Vue 2 in that respect. I could actually really use the help getting a Vue 3 proof of concept working instead, so we can start porting over features this version to that version, to help with people upgrading that want to keep their customized snapshots when switching to vue 3

**Comment by TheJaredWilcurt** [comment]
Take a look at the current PR #75 

Specifically these two files:

* [InlineFunctions.vue](https://github.com/tjw-lint/jest-serializer-vue-tjw/blob/70de07fcf2312045814b8e35cb4ae94d3767172b/tests/components/InlineFunctions.vue)
* [InlineFunctions.test.js.snap](https://github.com/tjw-lint/jest-serializer-vue-tjw/blob/70de07fcf2312045814b8e35cb4ae94d3767172b/tests/__snapshots__/InlineFunctions.test.js.snap)

let me know if that looks right to you

**Comment by TheJaredWilcurt** [comment]
@cameron-martin the Clear Inline Functions PR is ready to be merged. just wanted to see if you had any input before doing so.

**TLDR:**

this:
```js
<template>
  <div>
    <h1 :title="function () { return true; }">Classic</h1>
    <h2
      :title="function () {
        return true;
      }"
    >Multi-line</h2>
    <h3 :title="(x) => !x">Arrow</h3>
    <h4 :title="method">Reference</h4>
    <h5 :title="method()">Run</h5>
    <h6 title="function (a word used in programming) is typically followed by curly braces, ie {}">Red herring</h6>
  </div>
</template>

<script>
export default {
  name: 'InlineFunctions',
  methods: {
    method: function () {
      return 2;
    }
  }
};
</script>
```
gets this snapshot:
```html
<div>
  <h1 title="[function]">Classic</h1>
  <h2 title="[function]">Multi-line</h2>
  <h3 title="[function]">Arrow</h3>
  <h4 title="[function]">Reference</h4>
  <h5 title="2">Run</h5>
  <h6 title="function (a word used in programming) is typically followed by curly braces, ie {}">Red herring</h6>
</div>
```

**Comment by cameron-martin** [comment]
> @cameron-martin I'm not sure what you mean, If you want to do a proof-of-concept PR I can take a look. But I hesitate to spend any time doing Vnode specific stuff, since Vue 3 is so different from Vue 2 in that respect. I could actually really use the help getting a Vue 3 proof of concept working instead, so we can start porting over features this version to that version, to help with people upgrading that want to keep their customized snapshots when switching to vue 3

My latest thoughts on this are: instead of mutating the vnodes then passing it to a pre-built printer, maybe it wouldn't be so hard to create a custom printer that directly applied such transformations.

**Comment by TheJaredWilcurt** [comment]
Available in v3.17.0
